packages/dds/tree/src/tree/sparseTree.ts

```diff
@@ -0,0 +1,167 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { assert } from "@fluidframework/common-utils";
+import { FieldKey } from "../tree";
+import { topDownPath, UpPath } from "./pathTree";
+
+/**
+ * Sparse Tree of nodes.
+ *
+ * Contains both child and parent pointers, which are kept in sync.
+ *
+ * Each node is equivalent to a path through the tree.
+ * This tree structure stores a collection of these paths, but deduplicating the common prefixes of the tree
+ * prefix-tree style.
+ */
+export class SparseNode<TData> implements UpPath {
```

Comment:
Can this not be deduplicated with AnchorSet's PathNode? If not, we should at least remove all the references to AnchorSet in this file.

Reply:
It can be done but I decided not to do it because the AnchorSet PathNode code is likey to undergo some important changes soon as we merge our anchor and changeset code. I didn't want to make that work more complicated by introducing an abstraction layer, especially given that there's a good chance for the two implementations to diverge, or for this implementation being re-implemented on top of the AnchorSet abstraction (as opposed to the PathNode one).

I cleaned up the references to AnchorSet. Let me know if I missed any.